test/e2e_node/image_volume.go

```diff
@@ -56,13 +57,14 @@ var _ = SIGDescribe("ImageVolume", nodefeature.ImageVolume, func() {
 		e2eskipper.SkipUnlessFeatureGateEnabled(features.ImageVolume)
 	})
 
-	createPod := func(ctx context.Context, volumes []v1.Volume, volumeMounts []v1.VolumeMount) {
+	createPod := func(ctx context.Context, podName, nodeName string, volumes []v1.Volume, volumeMounts []v1.VolumeMount) {
 		pod := &v1.Pod{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      podName,
 				Namespace: f.Namespace.Name,
 			},
 			Spec: v1.PodSpec{
+				NodeName:      nodeName,
```

Comment:
This should be set only if nodeName is not empty, I guess.

Reply:
It is not a pointer, so setting it to "" is equivalent to not setting it.